tests/pytests/unit/utils/test_nacl.py

```diff
@@ -14,3 +50,122 @@ def test_fips_mode():
     with patch("salt.utils.nacl.__opts__", opts, create=True):
         ret = salt.utils.nacl.__virtual__()
         assert ret == (False, "nacl utils not available in FIPS mode")
+
+
+def test_keygen_sk_file(test_keygen):
+    """
+    test nacl.keygen function
+    with sk_file set
+    """
+    with temp_file("test_keygen_sk_file") as fpath:
+        with salt.utils.files.fopen(fpath, "wb") as wfh:
+            wfh.write(test_keygen["sk"])
+
+        # test sk_file
+        ret = nacl.keygen(sk_file=fpath)
+        assert "saved pk_file: {}.pub".format(fpath) == ret
+        salt.utils.files.remove(str(fpath) + ".pub")
+
+
+def test_keygen_keyfile(test_keygen):
+    """
+    test nacl.keygen function
+    with keyfile set
+    """
+    with temp_file("test_keygen_keyfile") as fpath:
```

Comment:
```suggestion
    with temp_file("test_keygen_keyfile") as fpath:
```
All `temp_file` calls should be using the form of `pytest.helpers.temp_file`

Reply:
so why not add it to suggestion as above, then I just have to hit the 'Commit suggestion' button :)

Reply:
Because it was too much to do it on GH, and vim replace would be faster :)

Reply:
:1st_place_medal:

Reply:
done